features/step_definitions/test_steps.rb

```diff
@@ -227,4 +227,21 @@
 
 Then(/^I print all the text contents in the page using html tag$/) do
   puts browser.html
+end
+
+Then (/^I hover over the text "([^"]*)"$/) do |arg|
+  browser.span(:text => arg).when_present.hover
+end
+
+Then (/^I print the subcategories of class "([^"]*)"$/) do |classname|
+  browser.div(:class => classname).spans(:class => "nav-text").each do |span|
```

Comment:
There is another step defined for only 
`browser.spans(:class => classname).each do |span|
puts span.text`  like you said. 
But the one with div-class, gives better results without any spaces in the printing part if u notice in the results.  So I'm not going to remove that until i find a better alternative.
Also,  I will try to make the gherkin sentence more generic in the next commit